Revit_Core_Engine/Query/OpeningSurface.cs

```diff
@@ -54,12 +63,102 @@ public static ISurface OpeningSurface(this FamilyInstance familyInstance, HostOb
             {
                 BoundingBoxIntersectsFilter bbif = new BoundingBoxIntersectsFilter(new Outline(bbox.Min, bbox.Max));
                 hosts = new FilteredElementCollector(familyInstance.Document).OfClass(typeof(HostObject)).WherePasses(bbif).Cast<HostObject>().ToList();
-                hosts = hosts.Where(x => x.FindInserts(true, true, true, true).Any(y => y.IntegerValue == familyInstance.Id.IntegerValue)).ToList();
+                hosts = hosts.Where(x => x.FindInserts(true, true, true, true).Any(y => (y.IntegerValue == familyInstance.Id.IntegerValue)|| (y.IntegerValue == parentId))).ToList();
             }
 
             List<ISurface> surfaces = new List<ISurface>();
 
-            if (hosts.Count == 0)
+            if (parentElem != null) // Indicates element is a nested instance and needs different approach to extract location
+            {
+                Document doc = familyInstance.Document;
+                Transaction t = new Transaction(doc);
+
+                FailureHandlingOptions failureHandlingOptions = t.GetFailureHandlingOptions().SetClearAfterRollback(true);
+                t.Start("Temp Delete Inserts");
+                List<ElementId> inserts = hosts.SelectMany(x => x.FindInserts(true, true, true, true)).Distinct().Where(x => x.IntegerValue != -1).ToList();
+                ParameterSet instanceParams = familyInstance.Parameters;
+                List<RevitParameter> bhomParams = new List<RevitParameter>();
+                foreach (Parameter parameter in instanceParams)
+                {
+                    RevitParameter bhomParam = Convert.ParameterFromRevit(parameter);
+                    bhomParams.Add(bhomParam);
+                }
+                    
+                LocationPoint locPt = familyInstance.Location as LocationPoint;
+                Element dummyInstance = doc.Create.NewFamilyInstance(locPt.Point, familyInstance.Symbol, hosts.First() as Element, Autodesk.Revit.DB.Structure.StructuralType.NonStructural);
```

Comment:
Why do you loop over all hosts if you place `dummyInstance` on the first? By the looks of code, you could use only it further down the line.

Reply:
The usage of the first host instance here is just for placing the dummy instance. The later usage of all hosts is to extract the total opening area resulting from the dummy instances cutout on all hosts in case it overlaps multiple hosts.